match_minima.py

```diff
@@ -0,0 +1,787 @@
+    letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+    rpt = int((len(minimaE[0]) / 26) + 1)
+    xlabs = [''.join(i)
+             for i in itertools.product(letters, repeat=rpt)][:ref_nconfs]
+    # xticks by numbers instead
+    #xlabs = range(len(minimaE[0]))
+
+    # create figure
+    fig = plt.figure(figsize=(20, 10))
+    ax = fig.gca()
+    ax.set_xticks(np.arange(-1, ref_nconfs + 1, 2))
+
+    # label figure; label xticks before plotting for better spacing.
+    plt.title(plttitle, fontsize=16)
+    plt.ylabel(ylabel, fontsize=20)
+    plt.xlabel("conformer minimum", fontsize=20)
+    plt.xticks(list(range(ref_nconfs)), xlabs, fontsize=18)
+    plt.yticks(fontsize=18)
+
+    # define line colors and markers
+    colors = mpl.cm.rainbow(np.linspace(0, 1, num_files))
+    markers = [
+        "x", "^", "8", "d", "o", "s", "*", "p", "v", "<", "D", "+", ">", "."
+    ] * 10
+
+    # plot the data
+    for i, file_ene in enumerate(minimaE):
+
+        # skip the non-selected ones if defined
+        if selected is not None and i not in selected:
+            continue
+
+        # define x's from integer range with step 1
+        xi = list(range(ref_nconfs))
+
+        # generate plot
+        plt.plot(xi, file_ene, color=colors[i], label=legend[i],
+            marker=markers[i], markersize=9)
+
+    # add legend, set plot limits, add grid
+    plt.legend(bbox_to_anchor=(0.96, 1), loc=2, prop={'size': 18})
+    plt.xlim(-1, ref_nconfs + 1)
+    ax.set_yticks(
+        np.arange(int(round(floor)) - 2,
+                  int(round(ceiling)) + 2))
+    plt.grid()
+
+    # save and close figure
+    plt.savefig(figname, bbox_inches='tight')
+    #plt.show()
+    fig.clear()
+    plt.close(fig)
+
+
+def match_minima(in_dict, rmsd_cutoff):
+    """
+    For different methods, match the conformer minima to those of the reference
+        method. Ex. Conf G of reference method matches with conf R of method 2.
+
+    Parameters
+    ----------
+    in_dict : OrderedDict
+        dictionary from input file, where key is method and value is dictionary
+        first entry should be reference method
+        in sub-dictionary, keys are 'sdfile' and 'sdtag'
+    rmsd_cutoff : float
+        cutoff above which two structures are considered diff conformers
+
+    Returns
+    -------
+    mol_dict : dict of dicts
+        mol_dict['mol_name']['energies']
+        = [[file1: conf1 conf2] [file2: conf1 conf2]]
+
+    """
+
+    # nested dictionary: 1st layer of mol names, 2nd layer of method energies
+    mol_dict = {}
+
+    # get first filename representing the reference geometries
+    sdf_ref = list(in_dict.values())[0]['sdfile']
+
+    # assess each file against reference
+    for ff_label, ff_dict in in_dict.items():
+        sdf_query = ff_dict['sdfile']
+        sdf_tag = ff_dict['sdtag']
+
+        # load molecules from open reference and query files
+        print("\n\nOpening reference file %s" % sdf_ref)
```

Comment:
Since you use f-strings elsewhere in this file, I'd recommend using them here for consistency